WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -538,21 +537,20 @@ class WeeklyRoundupNotificationScheduler {
         comments: Int,
         likes: Int,
         periodEndDate: Date,
-        context: NSManagedObjectContext,
-        completion: @escaping (Result<Void, Error>) -> Void) {
-
-            var siteTitle: String?
-            var dotComID: Int?
-
-            context.performAndWait {
-                dotComID = site.dotComID?.intValue
-                siteTitle = site.title
-            }
+        completion: @escaping (Result<Void, Error>) -> Void
+    ) {
+        var siteTitle: String?
+        var dotComID: Int?
+
+        site.managedObjectContext?.performAndWait {
+            siteTitle = site.title
+            dotComID = site.dotComID?.intValue
+        }
 
-            guard let dotComID = dotComID else {
-                // Error
-                return
-            }
+        guard let dotComID = dotComID else {
+            // Error
+            return
```

Comment:
Would it be useful to call the completion and pass an error about this to it?

Reply:
I don't think this "else case" can actually happen, not in the current code at least. The non-WP.com sites have been filtered out beforehand in 
https://github.com/wordpress-mobile/WordPress-iOS/blob/e3c3e41637a5319dd2577a3bba9be88a1ed6349a/WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift#L82-L85

Reply:
Gotcha. 

🤔 😞 It's disappointing the domain modeling evolved to leave the door open for this inconsistent state path, but I'd say tackling it is out scope for this PR.